syntax = "proto3";
package mongo;




import "google/protobuf/struct.proto";

/*
SearchAll请求
*/
message SearchAllRequest {
    /*
    表名
    */
    string collection = 1 ;
    /*
    查询条件
    */
    google.protobuf.Struct query = 2 ;
    /*
    fields
    */
    repeated string fields = 3 ;
    /*
    sort
    */
    repeated string sort = 4 ;
}

/*
SearchAll返回
*/
message SearchAllResponse {
    /*
    记录列表
    */
    repeated google.protobuf.Struct docList = 1 ;
}

/*
SearchAllApi返回
*/
message SearchAllResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    mongo.SearchAllResponse data = 4 ;
}

